app/src/main/java/com/github/se/polyfit/ui/screen/NutritionScreen.kt

```diff
@@ -79,14 +81,26 @@ private fun TopBar(navigateBack: () -> Unit) {
 }
 
 @Composable
-private fun BottomBar(setMeal: () -> Unit, isComplete: Boolean, navigateForward: () -> Unit) {
+private fun BottomBar(
+    setMeal: () -> Unit,
+    isComplete: Boolean,
+    navigateForward: () -> Unit,
+    updateMealData: (createdAt: LocalDate) -> Unit
+) {
   BottomAppBar(
       modifier = Modifier.height(128.dp).testTag("BottomBar"), containerColor = Color.Transparent) {
         Column(
             modifier = Modifier.fillMaxWidth().testTag("ButtonColumn"),
             horizontalAlignment = Alignment.CenterHorizontally) {
               PrimaryButton(
-                  onClick = { Log.v("Add Recipe", "Clicked") },
+                  onClick = {
+                    Log.v("Add Recipe", "Clicked")
+                    navigateForward()
+                    // Set the date of the meal to the minimum value to set a default value
+                    // who will not be on the data we see.
+                    updateMealData(LocalDate.MIN)
+                    setMeal()
```

Comment:
## Issue Description

There appears to be a problem when the meal is incomplete. The button `Add To Diary` remains always active even when not all the necessary information has been entered.. This issue allows the `setMeal()` function to be called prematurely, which leads to an error. The function is designed to check if the meal information is complete before it proceeds to store the data in the database.

### Example Of the Error
[Watch the example video](https://github.com/swent-group10/polyfit/assets/100274930/9966ae59-0916-4912-a4ff-19c67d88999d)